### PR TITLE
Setting up a small set of tests to run from mmtk-core's CI

### PR DESCRIPTION
**Pull request by udesou**

### CODE REVIEW

**Review by qinsoon — APPROVED** [review]
LGTM. The test passed in https://github.com/mmtk/mmtk-core/pull/941 in roughly 1 hour.

**Comment by udesou** [comment]
> LGTM. The test passed in [mmtk/mmtk-core#941](https://github.com/mmtk/mmtk-core/pull/941) in roughly 1 hour.

I've changed it a bit. I noticed that there's a rebuild after patching the tests. If we do the patching before the build, it should bring the time down further.

**Comment by qinsoon** [comment]
> > LGTM. The test passed in [mmtk/mmtk-core#941](https://github.com/mmtk/mmtk-core/pull/941) in roughly 1 hour.
> 
> I've changed it a bit. I noticed that there's a rebuild after patching the tests. If we do the patching before the build, it should bring the time down further as we only build it once.

Yeah. I noticed that as well. For CI in mmtk-julia, it seems we also do a rebuild.

**Comment by udesou** [comment]
I've created a subitem to apply the patches before the build for the CI in mmtk-julia as well.

**Comment by qinsoon** [comment]
With the new changes, the script failed here: https://github.com/mmtk/mmtk-core/actions/runs/6079604369/job/16492333680?pr=941

File: .github/scripts/ci-test-gc-core.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+set -e
+
+# run tests that are more specific to the GC
+# this list was built intuitively based on tests that 
+# call the gc manually, have finalizers, have multiple threads, etc.
+
+. $(dirname "$0")/common.sh
```
Comment (udesou) [inline .github/scripts/ci-test-gc-core.sh:7]: Are you sure it picked up the latest changes? I've added this line which should fix that error. Also note that the log points at line 25, which is now at line 27 after this change.

Comment (qinsoon) [reply .github/scripts/ci-test-gc-core.sh:7]: Probably I was running on an old commit. I will rerun it.

Comment (qinsoon) [reply .github/scripts/ci-test-gc-core.sh:7]: A rerun from mmtk-core passed: https://github.com/mmtk/mmtk-core/actions/runs/6079604369/job/16494059161?pr=941

**Pull request merged**